### ISSUE TITLE
Migrations.json networks field empty

### ISSUE THREAD

**Issue by honestbonsai** [trufflesuite/truffle#1738]
## Issue

`truffle migrate` doesn’t seem to write anything to the `networks` field in `Migrations.json`. is that a bug or is `Migrations.json` treated specially apart from other deployed contracts?

Related ganache issue https://github.com/trufflesuite/ganache/issues/1128#issuecomment-466143182

I could try looking into this if someone can point me in the right direction.

## Steps to Reproduce

Same as trufflesuite/ganache#1128

## Expected Behavior

`Migrations.json` `network` field shouldn't be empty. The proper address shows up in the terminal window. I'd expect the `network` field to reflect the same info.

## Actual Results

`network` field is empty

## Environment

* Operating System: Mac Mojave 10.14.2
* Ethereum client:
* Truffle version (`truffle version`): 5.0.0
* node version (`node --version`): 10.9.0
* npm version (`npm --version`): 6.4.1

**Comment by honestbonsai** [comment]
If you log out `resolve.contracts()` there's always an empty Migrations object at the end of the array

```
Starting migrations...
======================
> Network name:    'ganache'
> Network id:      5777
> Block gas limit: 6721975


1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   > transaction hash:    0xed940c9f801389a1b8676ef45199ab52467befda3a0bf8c410c8be6e04c69515
   > Blocks: 0            Seconds: 0
   > contract address:    0xb7e5aDA767458702B9cCdeE90569871B4e3c9025
   > account:             0x9827CfC78824baDa5c1A4802197fC5646DD8F5E3
   > balance:             99.11392156
   > gas used:            284908
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00569816 ETH

   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00569816 ETH

0 'Migrations' { '5777':
   { events: {},
     links: {},
     address: '0xb7e5aDA767458702B9cCdeE90569871B4e3c9025',
     transactionHash:
      '0xed940c9f801389a1b8676ef45199ab52467befda3a0bf8c410c8be6e04c69515' } }
1 'Migrations' {}

2_deploy_contracts.js
=====================

   Replacing 'SimpleStorage'
   -------------------------
   > transaction hash:    0xe0c27084b279898375c9e756951b9101adbb795bd0490b0905c505550df3a777
   > Blocks: 0            Seconds: 0
   > contract address:    0xcED2B31b5D4537F7774470b0F9b1f6c05e759109
   > account:             0x9827CfC78824baDa5c1A4802197fC5646DD8F5E3
   > balance:             99.1110313
   > gas used:            144513
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00289026 ETH


   Replacing 'TutorialToken'
   -------------------------
   > transaction hash:    0x8ece6ae1e9aac15de1051c0c24ff086698b9d9d9ce63d82ddc211cd577d1a064
   > Blocks: 0            Seconds: 0
   > contract address:    0x49f48e41a6C6b05aa82405a7786b45b6aFa297B1
   > account:             0x9827CfC78824baDa5c1A4802197fC5646DD8F5E3
   > balance:             99.08234308
   > gas used:            1434411
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.02868822 ETH


   Replacing 'ComplexStorage'
   --------------------------
   > transaction hash:    0x5f3a4649ba17f441b3b0ac46c13a28449a9d422617a2213d67de161b9f56bc9c
   > Blocks: 0            Seconds: 0
   > contract address:    0xef8756ffCEAc230B8B923d033C5b688155d4442B
   > account:             0x9827CfC78824baDa5c1A4802197fC5646DD8F5E3
   > balance:             99.05233648
   > gas used:            1500330
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.0300066 ETH

   > Saving artifacts
   -------------------------------------
   > Total cost:          0.06158508 ETH


Summary
=======
> Total deployments:   4
> Final cost:          0.06728324 ETH

0 'SimpleStorage' { '5777':
   { events: {},
     links: {},
     address: '0xcED2B31b5D4537F7774470b0F9b1f6c05e759109',
     transactionHash:
      '0xe0c27084b279898375c9e756951b9101adbb795bd0490b0905c505550df3a777' } }
1 'TutorialToken' { '5777':
   { events:
      { '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef': [Object],
        '0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925': [Object] },
     links: {},
     address: '0x49f48e41a6C6b05aa82405a7786b45b6aFa297B1',
     transactionHash:
      '0x8ece6ae1e9aac15de1051c0c24ff086698b9d9d9ce63d82ddc211cd577d1a064' } }
2 'ComplexStorage' { '5777':
   { events: {},
     links: {},
     address: '0xef8756ffCEAc230B8B923d033C5b688155d4442B',
     transactionHash:
      '0x5f3a4649ba17f441b3b0ac46c13a28449a9d422617a2213d67de161b9f56bc9c' } }
3 'Migrations' {}
```

**Comment by honestbonsai** [comment]
Ok I think I solved it, will submit a PR

**Comment by honestbonsai** [comment]
Solved by #1763

**Comment by ferittuncer** [comment]
This issue appears again with Truffle v5.0.35
